mobile_config/tests/gateway_service_v3.rs

```diff
@@ -0,0 +1,423 @@
+use chrono::{Duration, Utc};
+use futures::stream::StreamExt;
+use std::vec;
+
+use helium_crypto::{Keypair, PublicKey, Sign};
+use helium_proto::services::mobile_config::{
+    self as proto, DeploymentInfo, DeviceTypeV2, GatewayClient, GatewayInfoStreamReqV3,
+    GatewayInfoV3, LocationInfo,
+};
+use prost::Message;
+use sqlx::PgPool;
+
+pub mod common;
+use common::*;
+
+#[sqlx::test]
+async fn gateway_stream_info_v3_basic(pool: PgPool) {
+    let admin_key = make_keypair();
+    let asset1_pubkey = make_keypair().public_key().clone();
+    let asset1_hex_idx = 631711281837647359_i64;
+    let asset2_hex_idx = 631711286145955327_i64;
+    let asset2_pubkey = make_keypair().public_key().clone();
+    let now = Utc::now();
+    let now_plus_10 = now + chrono::Duration::seconds(10);
+    let now_plus_5 = now + chrono::Duration::seconds(5);
+
+    create_db_tables(&pool).await;
+    add_db_record(
+        &pool,
+        "asset1",
+        Some(asset1_hex_idx),
+        "\"wifiIndoor\"",
+        asset1_pubkey.clone().into(),
+        now,
+        Some(now),
+        Some(r#"{"wifiInfoV0": {"antenna": 18, "azimuth": 161, "elevation": 2, "electricalDownTilt": 3, "mechanicalDownTilt": 4}}"#)
+    )
+    .await;
+    add_mobile_tracker_record(
+        &pool,
+        asset1_pubkey.clone().into(),
+        now_plus_10,
+        Some(asset1_hex_idx),
+        Some(now_plus_5),
+    )
+    .await;
+
+    add_db_record(
+        &pool,
+        "asset2",
+        Some(asset2_hex_idx),
+        "\"wifiDataOnly\"",
+        asset2_pubkey.clone().into(),
+        now_plus_10,
+        Some(now_plus_10),
+        None,
+    )
+    .await;
+    add_mobile_tracker_record(&pool, asset2_pubkey.clone().into(), now_plus_10, None, None).await;
+
+    let (addr, _handle) = spawn_gateway_service(pool.clone(), admin_key.public_key().clone()).await;
+    let mut client = GatewayClient::connect(addr).await.unwrap();
+
+    // Select all devices
+    let req = make_gateway_stream_signed_req_v3(&admin_key, &[], 0, 0);
+    let mut stream = client.info_stream_v3(req).await.unwrap().into_inner();
+    let resp = stream.next().await.unwrap().unwrap();
+    assert_eq!(resp.gateways.len(), 2);
+
+    // Filter by device type
+    let req = make_gateway_stream_signed_req_v3(&admin_key, &[DeviceTypeV2::Indoor], 0, 0);
+    let mut stream = client.info_stream_v3(req).await.unwrap().into_inner();
+    let resp = stream.next().await.unwrap().unwrap();
+    assert_eq!(resp.gateways.len(), 1);
+    let gateway: &GatewayInfoV3 = resp.gateways.first().unwrap();
+    assert_eq!(gateway.device_type, Into::<i32>::into(DeviceTypeV2::Indoor));
```

Comment:
Rust structs with enums provide a method that returns their reified enum variant.
```suggestion
    assert_eq!(gateway.device_type(), DeviceTypeV2::Indoor);
```